### ISSUE TITLE
event pols 

### ISSUE THREAD

**Issue by DelinQu** [knelk/enerf#3]
Hi [knelk](https://github.com/knelk),

Thanks for your help in the previous issue. I still have a question about your code regarding the event sampling in loss calculation. Specifically, when calculating the event loss, you use the event polarity that has not been accumulated. Since two frames of pose-rendered images are used in computing the loss, why are the corresponding events sampled randomly instead of being accumulated?  

https://github.com/knelk/enerf/blob/3fb17cde79bb1c1967b3560d1a52bca6ab9168f5/nerf/provider.py#L1400C1-L1405C61

I have checked the configuration files to make sure that accumulate_evs is **False**.

I would greatly appreciate it if you could clarify this. Thank you for your time and help.

Best regards, 
Delin

**Comment by knelk** [comment]
Hi @DelinQu,

our default loss is without event accumulation. 

> Since two frames of pose-rendered images are used in computing the loss, why are the corresponding events sampled randomly instead of being accumulated?

We do not use two frames of pose-rendered images as loss. By default, we use the random pairs as loss of successor events.  Please refer to our paper III. 

![image](https://github.com/knelk/enerf/assets/69507924/f46d0589-1413-45e8-b0e1-7e1492136e1a)


In our ablation study, we also show the accumulated events loss. This loss is **slightly** worse in simulation, my hypothesis is because it only considers global windows at once, not the local high-freq details, so maybe the Nerf can pick up high-freq details quicker with our loss. 

However, I have thought a lot about this loss and I think in theory, if enough random accumulation windows are sampled, the NeRF should more or less converge to the same weights for either loss (accumulated or successor events). This is also what the experiments show (the values are very similar, maybe they just need slightly different hyperparameters due to numerical reasons etc.). On real data, sometimes the accumulated loss can sometimes be (slightly) better => again, I guess this is a function of the data / scene, which loss is better in the end, as sometimes you might want to only get a more smooth NeRF instead of overfitting the high-freq details and only indirectly learning the smoothness.


This could be an interesting further investigation for E-Nerf, especially in scenes with smooth vs. high-freq-texture. Maybe one needs to combine the two somehow, e.g. doing coarse-to-fine in the 3D volume space by first using the accumulated loss and later the successor-loss.

Hope it helps
